cmd/flux/debug_helmrelease.go

```diff
@@ -0,0 +1,122 @@
+	"github.com/fluxcd/pkg/apis/meta"
+	"github.com/fluxcd/pkg/chartutil"
+	"github.com/go-logr/logr"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/yaml"
+
+	"github.com/fluxcd/flux2/v2/internal/utils"
+)
+
+var debugHelmReleaseCmd = &cobra.Command{
+	Use:     "helmrelease [name]",
+	Aliases: []string{"hr"},
+	Short:   "Debug a HelmRelease resource",
+	Long: `The debug helmrelease command can be used to troubleshoot failing Helm release reconciliations.
+WARNING: This command will print sensitive information if Kubernetes Secrets are referenced in the HelmRelease .spec.valuesFrom field.`,
+	Example: `  # Print the status of a Helm release
+  flux debug hr podinfo --show-status
+
+  # Export the final values of a Helm release composed from referred ConfigMaps and Secrets
+  flux debug hr podinfo --show-values > values.yaml`,
+	RunE: debugHelmReleaseCmdRun,
+	Args: cobra.ExactArgs(1),
+}
+
+type debugHelmReleaseFlags struct {
+	name       string
+	showStatus bool
+	showValues bool
+}
+
+var debugHelmReleaseArgs debugHelmReleaseFlags
+
+func init() {
+	debugHelmReleaseCmd.Flags().BoolVar(&debugHelmReleaseArgs.showStatus, "show-status", false, "print the status of the Helm release")
+	debugHelmReleaseCmd.Flags().BoolVar(&debugHelmReleaseArgs.showValues, "show-values", false, "print the final values of the Helm release")
+	debugCmd.AddCommand(debugHelmReleaseCmd)
+}
+
+func debugHelmReleaseCmdRun(cmd *cobra.Command, args []string) error {
+	name := args[0]
+
+	if debugHelmReleaseArgs.showStatus == false && debugHelmReleaseArgs.showValues == false {
+		return fmt.Errorf("either --show-status or --show-values must be set")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), rootArgs.timeout)
+	defer cancel()
+
+	kubeClient, err := utils.KubeClient(kubeconfigArgs, kubeclientOptions)
+	if err != nil {
+		return err
+	}
+
+	hr := &helmv2.HelmRelease{}
+	hrName := types.NamespacedName{Namespace: *kubeconfigArgs.Namespace, Name: name}
+	if err := kubeClient.Get(ctx, hrName, hr); err != nil {
+		return err
+	}
+
+	if debugHelmReleaseArgs.showStatus {
+		status, err := yaml.Marshal(hr.Status)
+		if err != nil {
+			return err
+		}
+		rootCmd.Print(string(status))
+	}
+
+	if debugHelmReleaseArgs.showValues {
+		// TODO(stefan): remove the mapping when helm-controller/api v1.2.0 has been released
+		var valuesRefs []meta.ValuesReference
+		for _, source := range hr.Spec.ValuesFrom {
+			valuesRefs = append(valuesRefs, meta.ValuesReference{
+				Kind:      source.Kind,
+				Name:      source.Name,
+				ValuesKey: source.ValuesKey,
+				Optional:  source.Optional,
+			})
+		}
+
+		finalValues, err := chartutil.ChartValuesFromReferences(ctx,
+			logr.Discard(),
+			kubeClient,
+			hr.GetNamespace(),
+			hr.GetValues(),
+			valuesRefs...)
+		if err != nil {
+			return err
+		}
+
+		values, err := yaml.Marshal(finalValues)
+		if err != nil {
+			return err
+		}
+		rootCmd.Print(string(values))
+	}
+
+	return nil
```

Comment:
LGTM, just two comments:

* It would be nice to print something if neither flags are passed? Maybe default one of them to true, e.g. the status?
* If both flags are passed the two marshaled YAMLs will get merged together. Can we print the document separation between them if that happens? The `---`.

Reply:
> It would be nice to print something if neither flags are passed? 

An error is returned, see line 67. 

> Can we print the document separation 

Yeap, I'll do that, thanks!

Reply:
> An error is returned, see line 67.

Oh I missed that, sorry.